packages/tailwindcss-language-service/src/util/rewriting/replacements.ts

```diff
@@ -58,9 +77,20 @@ export function replaceCssVars(str: string, replace: CssVarReplacer): string {
           str = str.slice(0, i) + replacement + str.slice(j + 1)
         }
 
-        // We don't want to skip past anything here because `replacement`
-        // might contain more var(…) calls in which case `i` will already
-        // be pointing at the right spot to start looking for them
+        // Move the index back one so it can look at the spot again since it'll
+        // be incremented by the outer loop. However, since we're replacing
+        // variables recursively we might end up in a loop so we need to keep
+        // track of which variables we've already seen and where they were
+        // replaced to avoid infinite loops.
+        if (recursive) {
+          let key = `${i}:${replacement}`
+
+          if (!seen.has(key)) {
+            seen.add(key)
+            i -= 1
+          }
+        }
```

Comment:
@RobinMalfait or @philipp-spiess can y'all think of a way to break this? It has the potential for infinite recursion (due to circular theme key lookups) but I guarded against it here.

I think this should be sufficient but if you could give it some thought that would be much appreciated.